qiskit/circuit/library/generalized_gates/diagonal.py

```diff
@@ -71,21 +71,22 @@ class Diagonal(QuantumCircuit):
     `arXiv:0406176 <https://arxiv.org/pdf/quant-ph/0406176.pdf>`_
     """
 
-    def __init__(self, diag: list[complex] | np.ndarray) -> None:
+    def __init__(self, diag_dict: dict) -> None:
         """Create a new Diagonal circuit.
 
         Args:
-            diag: list of the 2^k diagonal entries (for a diagonal gate on k qubits).
+            diag: dict of the diagonal entries which are not 1.
 
         Raises:
-            CircuitError: if the list of the diagonal entries or the qubit list is in bad format;
-                if the number of diagonal entries is not 2^k, where k denotes the number of qubits
+            CircuitError: if the dictionary of the diagonal entries are in bad format;
+                          if absolute value of any entry is not 1.
```

Comment:
The indentation here was actually already correct - continuation lines just need to be indented by four spaces, not aligned with the colon.

Reply:
`pylint` fails due to word limit exceed in docstring hence kept in separate line.